### ISSUE TITLE
Encrypt communications

### ISSUE THREAD

**Issue by paralin** [FredKSchott/CoVim#81]
At the moment, from my quick glance at the code, it seems you're just blasting unencrypted JSON over a socket? I'd prefer to have some key xchange mechanism when first connecting to encrypt the session and prevent an unencrypted keylogger in vim :)

**Comment by ethanmad** [comment]
Yes please!

**Comment by FredKSchott** [comment]
Definitely, I haven't had the time to work on this project recently, but I'd gladly consider an encryption PR if someone were to write one.

**Comment by nicopace** [comment]
You may use Twisted own TLS support: https://twistedmatrix.com/documents/14.0.0/core/howto/ssl.html

**Comment by Manouchehri** [comment]
What about using SSH, similar to how mosh does their handshake?
